ddtrace/appsec/ddwaf/ddwaf_types.py

```diff
@@ -130,15 +128,18 @@ def __init__(self, struct=None):
             map_o = ddwaf_object_map(self)
             assert map_o
             # order is unspecified and could lead to problems if max_objects is reached
-            for (key, val) in struct.items():
+            for key, val in struct.items():
                 if not isinstance(key, (bytes, unicode)):  # discards non string keys
                     continue
                 res_key = key.encode("UTF-8", errors="ignore") if isinstance(key, unicode) else key
                 obj = ddwaf_object(val)
                 if obj.type:  # discards invalid objects
                     assert ddwaf_object_map_add(map_o, res_key, obj)
         else:
-            raise TypeError("ddwaf_object : unknown type in structure. " + repr(type(struct)))
+            if struct is not None:
+                log.warning("DDWAF object init called with unknown data structure")
+
+            ddwaf_object_invalid(self)
```

Comment:
is this safe? will everything else operate ok if a value gets converted to `None`?

Reply:
It should not happen, but if for any reason, an arbitratry object is used to feed the waf, it should count as invalid. This is an expected behaviour.

Reply:
cool, awesome. thanks for confirming. I'd always welcome a comment to explain that when I try to read this code again in the future 😆